testing/e2e/suites/operatorConfig.go

```diff
@@ -0,0 +1,96 @@
+package suites
+
+import (
+	"github.com/Azure/aks-app-routing-operator/testing/e2e/infra"
+	"github.com/Azure/aks-app-routing-operator/testing/e2e/manifests"
+)
+
+type cfgBuilder struct {
+	msi      string
+	tenantId string
+	location string
+}
+
+func builderFromInfra(infra infra.Provisioned) cfgBuilder {
+	return cfgBuilder{
+		msi:      infra.Cluster.GetClientId(),
+		tenantId: infra.TenantId,
+		location: infra.Cluster.GetLocation(),
+	}
+}
+
+type cfgBuilderWithOsm struct {
+	cfgBuilder
+	osmEnabled []bool
+}
+
+func (c cfgBuilder) withOsm(enabled ...bool) cfgBuilderWithOsm {
+	if len(enabled) == 0 {
+		enabled = []bool{false}
+	}
+
+	return cfgBuilderWithOsm{
+		cfgBuilder: c,
+		osmEnabled: enabled,
+	}
+}
+
+type cfgBuilderWithVersions struct {
+	cfgBuilderWithOsm
+	versions []manifests.OperatorVersion
+}
+
+func (c cfgBuilderWithOsm) withVersions(versions ...manifests.OperatorVersion) cfgBuilderWithVersions {
```

Comment:
Added an all manifests variable to get all. We don't have any exclusions on versions so far. I'll add that function when we get to that.

Not sure withExclusions is the right way. When we need them we'll need exclusions because the operator will behave in different ways. Maybe instead of having inclusions we have subsets of All like

`AllOperatorVersions`, `OperatorVersionsWithXFeature`, etc

Will think about it more when we get there